src/metrics/representation.py

```diff
@@ -146,228 +153,243 @@ def compute(self) -> float:
-            db_vecs: Database vectors, for faiss build index, size (database_size, embedding_size).
-            relevant: Array of arrays relevant indexes in database for every query request, size (queries_size, ).
-            scores: Array of scores related to queries that have at least one relevant item.
-            db_idxs: Array with all database indexes.
-            queries_idxs: Array of queries order number.
-        """
-        is_queries = queries_idxs >= 0
-        queries_idxs = queries_idxs[is_queries]
-        q_vecs = vectors[is_queries]
-        db_vecs = vectors[~is_queries]
-        db_idxs = np.arange(len(db_vecs))
-        scores = scores[~is_queries]
-        
-        relevant = []
-        empty_relevant_idxs = []
-
-        for idx in range(len(q_vecs)):
-            relevant_idxs = np.where(scores[:, queries_idxs[idx]] > 0.)[0]
-            if len(relevant_idxs) == 0:
-                empty_relevant_idxs.append(idx)
-            else:
-                # Need to sort relevant indexes by its scores for NDCG metric
-                current_scores = scores[relevant_idxs, queries_idxs[idx]]
-                sort_indexes = np.argsort(current_scores)
-                relevant_idxs = relevant_idxs[sort_indexes[::-1]]
-                relevant.append(relevant_idxs)
-        
-        relevant = np.array(relevant)
+            relevant_idxs: Array of relevant indexes in gallery data, for every query.
+            gallery_idxs: Array of gallery indexes in vectors storage.
+            query_column_idxs: Array of queries column indexes in scores matrix. It means query order number in origin
+                dataset, because after shuffle dataset the query order may be change we need to store it to reproduce
+                for every query vector it's relevant.
+            query_row_idxs: Array of queries row indexes in scores matrix, lso this index belong to vectors storage, 
+                need to get query vector in search study.
+            query_as_relevant: Array of query row indexes which are in relevant, i.e belong to queries and 
+                gallery simultaneously.
 
-        # remove empty relevant queries
-        q_vecs = np.delete(q_vecs, empty_relevant_idxs, axis=0)
-        queries_idxs = np.delete(queries_idxs, empty_relevant_idxs)
+        Raises:
+            ValueError: If dataset has query vector without relevants.
+        """
+        full_indexes = np.arange(len(query_idxs))
+        is_query = query_idxs >= 0
+        # array of columns indexes in scores matrix, need to get relevant
+        query_column_idxs = query_idxs[is_query]
+        # array of row indexes in scores matrix, also this index belong to vectors storage, need to get query vector in
+        # search study
+        query_row_idxs = full_indexes[is_query]
+        # gallery idxs in vectors storage, which row sum > 0
+        gallery_idxs = np.where(np.sum(np.abs(scores), axis=-1) > 0.)[0]
```

Comment:
I think this isn't a problem if you store queries in relevant. On search time you can exclude queries since they are always at top-1. The implementation will be simpler and equivalent for both types of datasets. The second argument is that doing this way `gallery_idxs` might exclude queries that are themselves relevant to other queries

Reply:
Actually i really don't know what will be faster compute `np.where(np.sum(np.abs(scores), axis=-1) > 0.)[0]` this operation or add query to gallery and search. I think it would be search because we do search many times. I think it's good idea to get this parameter from dataset to not compute this operation.

Reply:
There's also a way to calculate `np.any` by some axis